meta-ostro/conf/local.conf.sample

```diff
@@ -291,6 +291,18 @@ SUPPORTED_RECIPES_CHECK = "fatal"
 #
 #require conf/distro/include/ostro-os-production.inc
 
+# systemd-bootchart is a useful tool to analyze and optimize a system
+# boot time. The tool is available in Ostro and needs to be activated
+# by a kernel command-line parameter which requires to build a new
+# image. You also need to make sure that you enable CONFIG_SCHEDSTATS=y
+# (that will also set CONFIG_SCHED_INFO=y) in your kernel, this option
+# is not enabled by default. For instructions on how to modify an Ostro
+# kernel, please check doc/howtos/modifying-ostro-kernel.rst.
```

Comment:
The man page says systemd-bootchart is in /usr/lib, so init=/usr/lib/systemd/systemd-bootchart

Ulf

From: Geoffroy Van Cutsem [mailto:notifications@github.com]
Sent: Friday, June 10, 2016 6:59 AM
To: ostroproject/meta-ostro
Subject: Re: [ostroproject/meta-ostro] local.conf.sample: add option to enable systemd-bootchart (#204)


In meta-ostro/conf/local.conf.sample<https://github.com/ostroproject/meta-ostro/pull/204#discussion_r66617080>:

> @@ -291,6 +291,18 @@ SUPPORTED_RECIPES_CHECK = "fatal"

>  #

>  #require conf/distro/include/ostro-os-production.inc

>

> +# systemd-bootchart is a useful tool to analyze and optimize a system

> +# boot time. The tool is available in Ostro and needs to be activated

> +# by a kernel command-line parameter which requires to build a new

> +# image. You also need to make sure that you enable CONFIG_SCHEDSTATS=y

> +# (that will also set CONFIG_SCHED_INFO=y) in your kernel, this option

> +# is not enabled by default. For instructions on how to modify an Ostro

> +# kernel, please check doc/howtos/modifying-ostro-kernel.rst.

Here is what I have tried so far:

  *   Add a fragment file to my ostro-os repo gvancuts@iotdev:~/ostro-os$ cat meta-ostro-bsp/recipes-kernel/linux-yocto/linux-yocto/bootchart.cfg CONFIG_SCHEDSTATS=y
  *   Modify my local.conf: # Add a command-line parameter APPEND_append = " init=/lib/systemd/systemd-bootchart" SRC_URI_append_pn-linux-yocto = " file://bootchart.cfg<file:///\\bootchart.cfg>"

I have tried this for intel-quark so far (testing images in VirtualBox), the weird thing is that the kernel does have SCHEDSTAT enabled and the cmdline (as checked with 'less /proc/cmdline' on a running system has the extra init command - see below). However, there is no svg file created in /run/log

If I manually start /lib/systemd/systemd-bootchart, then such file is created as expected.

Any idea what is going wrong? Does the order of kernel command-line parameters matter?

root@intel-quark:~# cat /proc/cmdline
console=ttyS1,115200 console=tty0 init=/lib/systemd/systemd-bootchart fsck.mode=skip rootflags=i_version ro root=PARTUUID=12345678-9abc-def0-0fed-cba987654321 rootfstype=ext4

About the fact we have different kernel, I was thinking we could have multiple lines, one for each kernel (PN) in use in Ostro and with a simple note that should someone use yet a different kernel, they should adjust that very line. Does that sound like a good approach?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/ostroproject/meta-ostro/pull/204/files/918986fe9b4e4ec9d104c5fcd93e8c0e5db7172d#r66617080>, or mute the thread<https://github.com/notifications/unsubscribe/AA48t0MNJ8ekBYInE4rKb7l58t7HGAlAks5qKW2lgaJpZM4Iw_6M>.